pandas/core/generic.py

```diff
@@ -6815,16 +6815,21 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
                           errors=errors)
 
     _shared_docs['shift'] = ("""
-        Shift index by desired number of periods with an optional time freq
+        Shift index by desired number of periods with an optional time freq.
+
+        When freq is not passed, shift the index without realign the data.
+        If freq is passed (in this case, the index must be date or datetime),
+        the index will be increased using the periods and the freq.
 
         Parameters
         ----------
         periods : int
-            Number of periods to move, can be positive or negative
+            Number of periods to move, can be positive or negative.
         freq : DateOffset, timedelta, or time rule string, optional
             Increment to use from the tseries module or time rule (e.g. 'EOM').
             See Notes.
-        axis : %(axes_single_arg)s
+        axis : int or str
```

Comment:
The `axis` parameter has it's own standard (see the [guide](https://python-sprints.github.io/pandas/guide/pandas_docstring.html)):

```python
axis : {0 or ‘index’, 1 or ‘columns’, None}, default None
```